tools/snp.sh

```diff
@@ -1107,16 +1152,102 @@ attest_guest() {
 
   # Remove any special characters and print the value
   snpguest_report_measurement=$(echo ${snpguest_report_measurement} | sed $'s/[^[:print:]\t]//g')
+  echo -e "Measurement from SNP Attestation Report: ${snpguest_report_measurement}\n"
+
+	  # Compare the expected measurement to the guest report measurement
+	  [[ "${expected_measurement}" == "${snpguest_report_measurement}" ]] \
+		&& echo -e "The expected measurement matches the snp guest report measurement!" \
+		|| { >&2 echo -e "FAIL: measurements do not match"; return 1; }
+else 
+ 	 Verify_measurement
+fi
+}
+setup_svsm_guest_attestation() {
+
+   ssh_guest_command "
+   
+    # Update and install necessary packages
+    echo 'Updating package list...'
+    sudo apt-get update
+
+    echo 'Installing necessary packages...'
+    sudo apt-get install -y git build-essential libtss2-dev tpm2-tools
+
+    echo 'Installing Rust...'
+    source ""${HOME}"/.cargo/env" 2>/dev/null || true
+
+    curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs -sSf | sh -s -- -y
+    source ""${HOME}"/.cargo/env" 2>/dev/null
+    "    
+
+  # Check if 'snpguest' directory exists and remove it if it does
+  ssh_guest_command "
+  if [ -d \"snpguest\" ]; then
+     echo 'Directory snpguest exists. Removing it...'
+     rm -rf snpguest
+  else
+     echo 'Directory snpguest does not exist.'
+  fi
+"
+
+  # Clone the repository and build the project
+  ssh_guest_command "git clone https://github.com/virtee/snpguest.git"
+  ssh_guest_command "cd snpguest  &&  source \$HOME/.cargo/env &&  cargo build --release"
+}
```

Comment:
We already have logic for this, we decided we don't want to build the attestation tooling inside of the guest. We build it in the host and then scp it over to the guest. Look back at the snp guest attestation functions and build off those.

Reply:
The SVSM binary cannot be built on Ubuntu 22.04, which uses binutils version 2.35. SVSM requires binutils version 2.39 or higher (as documented in the Coconut-SVSM installation procedure), which is available on Ubuntu 24.04, the OS most commonly used by the community. so i have to build snpguest directly on guest VM and my function does same.

Reply:
I built the musl release of the snpguest  tool, updated the code, and verified it. The library compatibility issue is now resolved, and everything is working as expected and also removed setup_svsm_guest_attestation()